$OpenBSD: patch-rust_libnewsboat_src_utils_rs,v 1.1 2021/06/19 10:49:53 tb Exp $

iconv symbols are not part of libc on OpenBSD, so link agaist converters/libiconv

Index: rust/libnewsboat/src/utils.rs
--- rust/libnewsboat/src/utils.rs.orig
+++ rust/libnewsboat/src/utils.rs
@@ -892,10 +892,13 @@ type iconv_t = *mut c_void;
 // On FreeBSD, link with GNU libiconv; the iconv implementation in libc doesn't support //TRANSLIT
 // and WCHAR_T. This is also why we change the symbol names from "iconv" to "libiconv" below.
 #[cfg_attr(target_os = "freebsd", link(name = "iconv"))]
+#[cfg_attr(target_os = "openbsd", link(name = "iconv"))]
 extern "C" {
     #[cfg_attr(target_os = "freebsd", link_name = "libiconv_open")]
+    #[cfg_attr(target_os = "openbsd", link_name = "libiconv_open")]
     pub fn iconv_open(tocode: *const c_char, fromcode: *const c_char) -> iconv_t;
     #[cfg_attr(target_os = "freebsd", link_name = "libiconv")]
+    #[cfg_attr(target_os = "openbsd", link_name = "libiconv")]
     pub fn iconv(
         cd: iconv_t,
         inbuf: *mut *mut c_char,
@@ -904,6 +907,7 @@ extern "C" {
         outbytesleft: *mut size_t,
     ) -> size_t;
     #[cfg_attr(target_os = "freebsd", link_name = "libiconv_close")]
+    #[cfg_attr(target_os = "openbsd", link_name = "libiconv_close")]
     pub fn iconv_close(cd: iconv_t) -> c_int;
 }
 
